HelgaShiryaeva/assignment2/BinaryTree.java

```diff
@@ -1,104 +1,166 @@
 package assignment2;
 
 import java.util.HashSet;
-import java.util.Optional;
+import java.util.NoSuchElementException;
 import java.util.Set;
 
 /**
  * This class represents a Binary tree data structure
  * @param <T> is a type of values stored in nodes of a tree
+ * All values stored in a binary tree are unique

```

Comment:
This should go before params list. Typical structure of comment is:

- Summary.
- More details and assumptions.
- Params.